linera-service/tests/end_to_end_tests.rs

```diff
@@ -1042,40 +1094,95 @@ impl Application {
             assert_eq!(value, amount);
         }
     }
+}
+
+struct CounterApplication {
+    application: Application,
+}
+
+#[cfg(any(feature = "wasmer", feature = "wasmtime"))]
+impl CounterApplication {
+    async fn query_application(&self, query: &str) -> Value {
```

Comment:
I'd remove the general `query_application` from all the specific application types then, and replace it with a function for each query that we use in the tests.